client/src/components/layout/NewDogForm.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react"
+import { Redirect } from "react-router-dom";
+
+import Fetcher from "./services/Fetcher.js";
+import ErrorList from "./ErrorList.js"
+
+const NewDogForm = ({ user }) => {
+  const defaultFormValue = {
+    dogName: "",
+    description: ""
+  };
+
+  const [newDog, setNewDog] = useState(defaultFormValue);
+  const [errors, setErrors] = useState([]);
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+
+  const getNewDog = async () => {
+    const response = await Fetcher.post(`/api/v1/dogs/new`, newDog);
+    if (response.ok) {
+      return setShouldRedirect(true);
+    }
+    setErrors(response.validationErrors);
+  };
+
+  const handleInputChange = (event) => {
+    const { name, value } = event.currentTarget;
+    setNewDog({
+      ...newDog,
+      [name]: value,
+    });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    getNewDog();
+  };
+
+  const clearForm = () => {
+    setNewDog(defaultFormValue);
+  };
+
+  if (shouldRedirect) {
+    return <Redirect push to="/" />;
+  }
+
+  return (
+    <div className="grid-container">
```

Comment:
`grid-container` is a great class to put on a higher up element so that all pages can benefit from it and it doesn't need to be applied in every component